src/main/java/no/nav/pto/veilarbportefolje/arenaaktiviteter/TiltakRepositoryV2.java

```diff
@@ -0,0 +1,219 @@
+                        .collect(toMap(Tiltakkodeverk::getKode, Tiltakkodeverk::getVerdi))
+        );
+    }
+
+    public List<AktorId> hentBrukereMedUtlopteTiltak() {
+        String sql = "SELECT "+AKTOERID+" FROM " + TABLE_NAME
+                + " WHERE " + TILDATO + " < CURRENT_TIMESTAMP";
+        return db.queryForList(sql, String.class).stream()
+                .filter(Objects::nonNull).map(AktorId::new).collect(toList());
+    }
+
+    public List<Timestamp> hentSluttdatoer(PersonId personId) {
+        if (personId == null) {
+            throw new IllegalArgumentException("Trenger personId for å hente ut sluttdatoer");
+        }
+
+        final String hentSluttDatoerSql = "SELECT " + TILDATO + " FROM " + TABLE_NAME +
+                " WHERE "+PERSONID+"=?";
+        return db.queryForList(hentSluttDatoerSql, Timestamp.class, personId.getValue());
+    }
+
+
+    public List<Timestamp> hentStartDatoer(PersonId personId) {
+        if (personId == null) {
+            throw new IllegalArgumentException("Trenger personId for å hente ut startdatoer");
+        }
+
+        final String hentSluttDatoerSql = "SELECT " + FRADATO + " FROM " + TABLE_NAME +
+                " WHERE "+PERSONID+"=?";
+        return db.queryForList(hentSluttDatoerSql, Timestamp.class, personId.getValue());
+    }
+
+    /*
+    Kan forenkles til kun en bruker ved overgang til postgres
+    String sql = "SELECT DISTINCT " + Table.BRUKERTILTAK_V2.TILTAKSKODE + " FROM " + Table.BRUKERTILTAK_V2.TABLE_NAME
+            + " WHERE " + Table.BRUKERTILTAK_V2.AKTOERID + " = ?";
+     */
+    public Map<AktorId, Set<BrukertiltakV2>> hentBrukertiltak(List<AktorId> aktorIder) {
+        if (aktorIder == null || aktorIder.isEmpty()) {
+            throw new IllegalArgumentException("Trenger aktor-ider for å hente ut tiltak");
+        }
+        Map<String, Object> params = new HashMap<>();
+        params.put("aktorIder", aktorIder.stream().map(AktorId::toString).collect(toList()));
+
+        return namedParameterJdbcTemplate
+                .queryForList(getTiltaksTyperForListOfAktorIds(), params)
+                .stream()
+                .map(this::mapTilBrukertiltakV2)
+                .collect(toMap(BrukertiltakV2::getAktorId, DbUtils::toSet,
+                        (oldValue, newValue) -> {
+                            oldValue.addAll(newValue);
+                            return oldValue;
+                        }));
+    }
+
+    public void utledOgLagreTiltakInformasjon(PersonId personId, AktorId aktorId) {
+        List<BrukertiltakV2> tiltak = hentTiltak(aktorId);
+        LocalDate yesterday = LocalDate.now().minusDays(1);
+        Timestamp nesteUtlopsdato = tiltak.stream()
+                .map(BrukertiltakV2::getTildato)
+                .filter(Objects::nonNull)
+                .filter(utlopsdato -> utlopsdato.toLocalDateTime().toLocalDate().isAfter(yesterday))
+                .max(Comparator.naturalOrder())
```

Comment:
if its next expiration date, are we looking for nearest date to now, or furthest date from now?